### PR TITLE
ODATA-1488: Corrections to edm.xsd

### PR DESCRIPTION
**Pull request by HeikoTheissen**
The current XSD files use XSD version 1.0 and have limitations:
* They allow certain elements (e.g., `EntityType/Key`) to occur more often than the spec says.
* They do not allow `Annotation` elements in arbitrary places (e.g,. between two `Int` children of an `Add` element).
* Certain restrictions (e.g., on simultaneous `Int` attribute and child) cannot be expressed even with XSD version 1.1.

This PR adds a Java-based XML Schema validation test (based on XSD 1.0) to the repository.

### CODE REVIEW

File: examples/miscellaneous.xml [hunk 1]
```diff
@@ -63,6 +63,13 @@
     <!-- Example 6 -->
     <Schema Namespace="org.example" Alias="self">
       <Annotation Term="Core.Description" String="Example schema" />
+      <Annotation Term="Core.Version">
+        <Add>
+          <Int>1</Int>
+          <Annotation Term="Core.Description" String="add major and minor part" />
```
Comment (HeikoTheissen) [inline examples/miscellaneous.xml:69]: This `Annotation` between the first and second child of `Add` is not allowed by the `xs:sequence` given in https://github.com/oasis-tcs/odata-csdl-schemas/blob/e5b9fc7bc260bec8fdbba48fd4ddee1a84277981/schemas/edm.xsd#L420-L426

Comment (ralfhandl) [reply examples/miscellaneous.xml:69]: Which is not a problem for me if it is possible to put annotations before and after the operand expressions.

Comment (HeikoTheissen) [reply examples/miscellaneous.xml:69]: But the [OData-CSDL-XML] spec does not restrict the ordering of XML elements in any way.

File: java/Validator.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import javax.xml.validation.SchemaFactory;
+import javax.xml.XMLConstants;
+import java.io.File;
+import java.io.IOException;
+import org.xml.sax.SAXException;
+
+public class Validator {
+	public static void main(String[] args) throws IOException, SAXException {
+		var schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+		var validator = schemaFactory.newSchema(new StreamSource(new File("schemas/edmx.xsd"))).newValidator();
```
Comment (ralfhandl) [inline java/Validator.java:11]: Where is `edm.xsd` read from?

File: java/Validator.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+		validator.validate(new StreamSource(new File("examples/miscellaneous.xml")));
```
Comment (ralfhandl) [inline java/Validator.java:14]: Better loop over all files in `examples` folder

File: java/Validator.java (hunk 2 above)
Comment (HeikoTheissen) [reply java/Validator.java:18]: Have removed that, `edm.xsd` is now read automatically from the `xs:import` statement.

File: package.json [hunk 4]
```diff
@@ -30,7 +30,9 @@
   },
   "scripts": {
     "build": "node lib/transform.js",
-    "test": "c8 mocha"
+    "build-java": "javac java/Validator.java",
+    "test": "c8 mocha --reporter-option maxDiffSize=0",
+    "test-xsd": "java -cp java Validator"
```
Comment (ralfhandl) [inline package.json:35]: We could combine the two test steps:
```suggestion
    "test": "npm run test-js && npm run test-xsd",
    "test-js": "c8 mocha --reporter-option maxDiffSize=0",
    "test-xsd": "java -cp java Validator"
```
Same for the two build steps, then `nodejs.yml` doesn't have to be changed.

**Review by ralfhandl — APPROVED** [review]

**Pull request merged**